saleor/checkout/utils.py

```diff
@@ -258,25 +271,23 @@ def _append_line_to_update(
             to_update.append(line)
 
 
-def _append_line_to_delete(to_delete, variant, line_data, variant_ids_in_lines):
-    if variant.pk not in variant_ids_in_lines:
+def _append_line_to_delete(to_delete, line_data, line):
+    if line is None:
         return
-    line = variant_ids_in_lines[variant.pk]
+
     quantity = line_data.quantity
     if line_data.quantity_to_update:
         if quantity <= 0:
             to_delete.append(line)
 
 
-def _append_line_to_create(
-    to_create, checkout, variant, line_data, variant_ids_in_lines
-):
-    if variant.pk not in variant_ids_in_lines:
+def _append_line_to_create(to_create, checkout, variants_map, line_data, line):
```

Comment:
You use `variants_map` only in `variant=variants_map[line_data.variant_id],`. In that case, wouldn't be better to pass here `variant` instead of `variants_map`?

Reply:
What about this? Is there any reason for passing the whole variants_amp here instead of variant taken from the map?

Reply:
done, sorry i didnt pushed last commit